src/components/select/CustomSelectDropdown.tsx

```diff
@@ -77,6 +79,38 @@ const CustomSelectDropdown: FC<Props> = ({
     (searchable === "always" || (searchable === "auto" && options.length >= 5));
 
   useEffect(() => {
+    if (dropdownRef.current) {
+      const toggle = document.getElementById(toggleId);
+
+      // align width with wrapper toggle width
+      const toggleWidth = toggle?.getBoundingClientRect()?.width ?? 0;
+      dropdownRef.current.style.setProperty("min-width", `${toggleWidth}px`);
+
+      // align z-index: if in a modal context, take the next z-index we find above + 1
+      const getRecursiveZIndex = (element: HTMLElement | null): string => {
```

Comment:
I think this function can exist outside of the component

Reply:
moved it into a dedicated file, true this might be handy in other places.